### ISSUE TITLE
Reading notification

### ISSUE THREAD

**Issue by daniele-athome** [kontalk/androidclient#421]
Using some existing XEP if possible.

**Comment by webratte** [comment]
If I understand it right, it's the same thing like the blue checkmarks in WhatsApp!?

You will imlement it as a userpreference -switched off by default- I suppose :-P

**Comment by daniele-athome** [comment]
Yes you got it right and yes there will be a preference - but I think it will be on by default, otherwise people would not even know about it :-(

**Comment by webratte** [comment]
There was a big discussion after implement this feature in WhatsApp.

What do you think about ask user on first App start for enable/disable reading notification?
Of course if it's not too much work.

**Comment by giner** [comment]
I guess the main question with WhatsApp was related to inability to disable this behaviour.

**Comment by ameenross** [comment]
> What do you think about ask user on first App start for enable/disable reading notification?

That would probably need implementing a wizard (people tend to just press "next" until done) or a library such as [ShowcaseView](https://github.com/amlcurran/ShowcaseView)

**Comment by daniele-athome** [comment]
I think he's right: the problem was not being able to disable it.

I don't think it's such a big deal. I wanted to introduce some kind of showcase feature, but it's not going to highlight this feature I think.
A user preference under "privacy settings" will suffice. I'll improve documentation to make it clear that there is a whole settings section dedicated to privacy that people can tweak.

**Comment by webratte** [comment]
OK, I think your right.
Maybe I'm just a little bit paranoid ;-)

But I still think users should after starting  the App first time informed about this new feature.
In some Apps there is after update a simple window (with just a OK Button) shown "Whats new" (e.g. in K9 Mail).
And there are all changes since last update.
What do you think about that?

**Comment by daniele-athome** [comment]
Errrh... I've always been against those kind of dialogs. Those information should be and stay in the publishing platform. It's a shame F-Droid still doesn't have a "recent changes" box.

**Comment by webratte** [comment]
OK, you are the developer.
I will be quiet for now ;-)

But: 
I'LL BE BACK :-D

**Comment by daniele-athome** [comment]
Sorry :P you know I'm always open to suggestions, but I also like to keep things were they are supposed to be.

**Comment by h-2** [comment]
I am still in favor of per user settings that you see immediately in the invitation workflow, see also #300 (this part of the issue was not yet resolved).

**Comment by daniele-athome** [comment]
@h-2 unfortunately #300 is a big issue and I had to postpone it to a later release.
Release 3.0.1 will have an application-wide preference; #300 will introduce per-user settings for encryption and other privacy-related stuff.

**Comment by h-2** [comment]
Yeah, sure, no rush! Just wanted to mention it, since it was related to the current issue :)

**Comment by webratte** [comment]
@daniele-athome Will you fix the problem with the second checkmark (I'v not tested it yet with Version 3.0), we talked about in #416, before you implement reading notification?

**Comment by daniele-athome** [comment]
I don't know the exact timing... Tigase issues are coordinated with the Tigase team, it's not as fast as here...

**Comment by webratte** [comment]
Just a idea. Maybe it's hard (or impossible) to implement. But I still ask you ;-)

My idea is a option to enable/disable reading notification for each contact individually.
For example:
In global settings reading notifications are enabled.
But I can disable it for e.g. my boss .
Because my wife should know that I have read here message but my boss must not know that I have read his message.

**Comment by webratte** [comment]
Oops...
I have just seen @h-2 has mentioned it already with other words. Sorry for duplicate.

**Comment by TheLastProject** [comment]
Would [XEP-0184](https://xmpp.org/extensions/xep-0184.html) perhaps be what you want? [Chapter 7](https://xmpp.org/extensions/xep-0184.html#format) seems to talk about "[...] an ack message by a receiving entity that wishes to inform the sending entity that the content message has been received", which seems to be exactly what you want.

**Comment by daniele-athome** [comment]
XEP-0184 is what we currently use as delivery receipt (the double tick mark). What I'd like here is one further step: message received -> message read, which is not covered by XEP-0184.
